### PR TITLE
Hessian bug fix

### LINKED ISSUE #4: Hessian bug fix
Normally, MadNLP shouldn't regularize the problem for convex QPs, but it is regularizing this problem. This is an indication of a potential bug. This should be root-caused and fixed
https://github.com/MadNLP/DynamicNLPModels.jl/runs/6838395675?check_suite_focus=true#step:5:656

### PR DESCRIPTION
**Pull request by dlcole3**
Resolved the bug that resulted in regularizing a convex QP within `MadNLP.jl`. 

`LQDynamicModel` included a $u_N$ variable that was not needed within the optimization problem. Removed this variable by removing empty row/column in Hessian and empty column in Jacobian and extra lines in `lvar` and `uvar`. 

Also updated `build_QP_JuMP_model` function within `sparse_lq_test.jl` to remove the $u_N$ variable.

### CODE REVIEW

File: examples/JuMP_sparse_linear_mpc.jl [hunk 1]
```diff
@@ -0,0 +1,79 @@
+using Ipopt, JuMP, Random, LinearAlgebra
```
Comment (sshin23) [inline examples/JuMP_sparse_linear_mpc.jl]: these files should be removed

**Review by sshin23 — APPROVED** [review]
approve the changes, but remove the old files in `examples`

File: src/DynamicNLPModels.jl [hunk 2]
```diff
@@ -562,6 +562,55 @@ function _build_H(
 end
```
Comment (sshin23) [inline src/DynamicNLPModels.jl]: shouldn't appear yet

**Review by sshin23 — CHANGES_REQUESTED** [review]

File: src/DynamicNLPModels.jl (hunk 2 above)
Comment (dlcole3) [reply src/DynamicNLPModels.jl]: Sorry, I thought I had done my branch correctly to start the dense formulation, but that is not the case. I will remove that and push again

**Review by sshin23 — APPROVED** [review]

**Pull request merged**